hw/dv/tools/ralgen/ralgen.py

```diff
@@ -18,9 +19,22 @@
 except ImportError:
     from yaml import SafeLoader as YamlLoader
 
-# Repo root is 4 levels up. Note that this will require an update if the path to
-# this tool is changed.
-REPO_ROOT = "../../../.."
+
+# Get the REPO_TOP directory path. This is used to construct full paths below.
```

Comment:
Nit: This could be useful for other scripts, but it doesn't seem we have some kind of library to place it under.

Reply:
yes that's correct. I created an issue to track, see #13073. I can address this one in a follow up. I think there are other common functions we could include in such a library too.